### DISCUSSION TITLE
Is it possible to use a non literal (constant or parameter) value for Dialog.enable into Annotations?

### DISCUSSION THREAD

**Discussion by mpradovelasco** [OpenModelica/OpenModelica#10678] [Q&A] [answered]
This feature is really relevant for rich designs, since it helps to hide parameters and variables in the dialog window of a component as a function of the component's scope. I have seen many examples that use parameters or constants (for instance in the Modelica Language Specification) but it seems that it is not implemented in OpenModelica currently?

Thanks again!

**Comment by adeas31** [comment, marked as answer]
We do support this feature via the instance API. You need to check `Enable instance API` in ´Tools->Options´. We will make this default soon.

**Reply by mpradovelasco** [reply]
Thank you!! This feature solve many pending issues in my design.

**Comment by mpradovelasco** [comment]
Hello again, 

I have tested that option and I would like to comment some issues (I apologize if they are known):
 - Concerning the functionality, If I select enable = boolean (with value false), the parameter (for example) appears in the dialog window, but with a different state to components with enable = true (eg. it impedes to write a value to the parameter). I do not know if this is the final and selected behaviour. However, I expected a behaviour similar to current one (component does not appear in the dialog). 
 - Icons and diagrams with fillpatterns (mainly) generate scripting errors as "Error: Variable FillPattern.Solid not found in scope bryPortPBPK" that impede to see propertly the model diagram.
 - Although the model may be instatiated, many new warnings appear.

Thanks for your help!!

**Reply by casella** [reply]
> Hello again,
> 
> I have tested that option and I would like to comment some issues (I apologize if they are known):
> 
>     * Concerning the functionality, If I select enable = boolean (with value false), the parameter (for example) appears in the dialog window, but with a different state to components with enable = true (eg. it impedes to write a value to the parameter). I do not know if this is the final and selected behaviour. However, I expected a behaviour similar to current one (component does not appear in the dialog).

That's what the Modelica Language specification says in [Section 18.7](https://specification.modelica.org/maint/3.6/annotations.html#annotations-for-the-graphical-user-interface):

> If enable = false, the input field may be disabled and no input can be given.

Disabled doesn't mean invisible, it means you can't input a value.

>     * Icons and diagrams with fillpatterns (mainly) generate scripting errors as "Error: Variable FillPattern.Solid not found in scope bryPortPBPK" that impede to see propertly the model diagram.

This looks like a bug. Can you please create a simple test case that demonstrates the problem and open an issue about it?

>     * Although the model may be instatiated, many new warnings appear.

Same as above, if you want us to fix issues, you need to post test cases to reproduce them :)

**Reply by mpradovelasco** [reply]
You have full reason concerning the behaviour of enable according the language specification. I have commented it because until now, without selecting the `enable Instance API`, the input field does not appear when enable = false (in my opinion it is also compliant with the specification and simplifies the dialog interface), 

I will prepare a test case to reproduce the errors mentioned and open an issue as suggested. 

Thank you very much again for your help!

**Reply by adeas31** [reply]
Please also provide a test case for enable issue as well.

**Comment by mpradovelasco** [comment]
Hello, I have created a report in #10697 with details and a minimal test case. 

Thanks!
